### PR TITLE
Update to latest google-cloud-java and Cloud SQL

### LINKED ISSUE #711: Cloud SQL usage metrics not working

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #711, #718

### CODE REVIEW

**Comment by joaoandremartins** [comment]
As part of this PR, could you also remove the declaration of `google-api-client` in the root `pom.xml`?

File: spring-cloud-gcp-autoconfigure/pom.xml [hunk 1]
```diff
@@ -78,16 +78,6 @@
             <groupId>com.google.cloud.sql</groupId>
             <artifactId>mysql-socket-factory</artifactId>
             <optional>true</optional>
-            <exclusions>
-                <exclusion>
-                    <groupId>mysql</groupId>
```
Comment (joaoandremartins) [inline spring-cloud-gcp-autoconfigure/pom.xml]: We might still need this one?

File: spring-cloud-gcp-dependencies/pom.xml [hunk 2]
```diff
@@ -119,25 +120,13 @@
 			<dependency>
 				<groupId>com.google.cloud.sql</groupId>
 				<artifactId>mysql-socket-factory</artifactId>
-				<version>1.0.8</version>
-				<exclusions>
```
Comment (joaoandremartins) [inline spring-cloud-gcp-dependencies/pom.xml]: Is this library no longer pulled in by them?

File: spring-cloud-gcp-autoconfigure/pom.xml (hunk 1 above)
Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/pom.xml]: This dependency is now marked as `provided`. So, I don't think we need to exclude it.
https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/28b266b13ec4a9b6204226a864453d3903095f7e/connector-j-5/pom.xml#L25

File: spring-cloud-gcp-dependencies/pom.xml (hunk 2 above)
Comment (meltsufin) [reply spring-cloud-gcp-dependencies/pom.xml]: They exclude it: https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/28b266b13ec4a9b6204226a864453d3903095f7e/core/pom.xml#L29

File: spring-cloud-gcp-dependencies/pom.xml [hunk 3]
```diff
@@ -16,8 +16,9 @@
 	<packaging>pom</packaging>
 
 	<properties>
-		<google-cloud-bom.version>0.50.0-alpha</google-cloud-bom.version>
-		<cloud-trace-java.version>0.5.0</cloud-trace-java.version>
+		<google-cloud-bom.version>0.51.0-alpha</google-cloud-bom.version>
+		<google-auth.version>0.9.1</google-auth.version>
```
Comment (joaoandremartins) [inline spring-cloud-gcp-dependencies/pom.xml]: nit, but should we call this `google-auth-library-oauth2-http.version`, not to confuse with other Google auth libraries?

**Comment by joaoandremartins** [comment]
This also fixes #718

**Pull request merged**